tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -31,3 +43,105 @@ def test_sagemaker_handle_client_errors():
     except ClientError as e:
         result = _parse_aws_client_exception_or_raise(e)
     assert result.status_code == Status.UNAUTHENTICATED
+
+
+def test_cleanup_sagemaker_functions():
+    client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(client)
+    stubber.add_client_error(
+        method='delete_model', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_model(client, 'test_name', 'test_version')
```

Comment:
test the two cleanup functions separately as unit test, and add integration tests to the entire CreateDeployment implementation, make sure to cover cases like "model creation failed", "endpoint creation failed", "failed to cleanup model", "failed to cleanup endpoint" etc and verify the behavior

Reply:
done